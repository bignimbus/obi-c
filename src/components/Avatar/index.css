.avatar {
  height: 100%;
  position: relative;
  width: 100%;
}

.avatar__pixel {
  background-color: var(--theme-contrast-color--translucent);
  border: 1px solid var(--theme-contrast-color--translucent);
  position: absolute;
  transform: scale(1);
  transition: transform 50ms;
}

.avatar__pixel--short {
  transform: scale(1, 0.5);
}

.avatar__pixel--narrow {
  transform: scale(0.5, 1);
}
