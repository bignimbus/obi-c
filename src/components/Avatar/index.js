import './index.css';
import React, { useState } from 'react';

export const PIXELS_WIDE = 12;
export const PIXELS_TALL = 13;

const STILL_SPRITE = [
  [0, 0],
  [0, 1],
  [0, 2],
  [0, 3],
  [0, 4],
  [0, 5],
  [0, 6],
  [0, 7],
  [0, 8],
  [0, 9],
  [0, 10],
  [0, 11],
  [0, 12],
  [1, 12],
  [2, 12],
  [3, 12],
  [4, 12],
  [5, 12],
  [6, 12],
  [7, 12],
  [8, 12],
  [9, 12],
  [10, 12],
  [11, 12],
  [11, 0],
  [11, 1],
  [11, 2],
  [11, 3],
  [11, 4],
  [11, 5],
  [11, 6],
  [11, 7],
  [11, 8],
  [11, 9],
  [11, 10],
  [11, 11],
  [1, 0],
  [2, 0],
  [3, 0],
  [4, 0],
  [5, 0],
  [6, 0],
  [7, 0],
  [8, 0],
  [9, 0],
  [10, 0],
  [4, 4],
  [6, 4],
];

const Avatar = () => {
  const [sprite, setSprite] = useState(STILL_SPRITE);
  return (
    <div className='avatar'>
      {
        sprite.map(([x, y], i) => (
          <div
            key={i}
            className='avatar__pixel'
            style={{
              left: `calc(100% / ${PIXELS_WIDE} * ${x}`,
              height: `calc(100% / ${PIXELS_TALL})`,
              top: `calc(100% / ${PIXELS_TALL} * ${y})`,
              width: `calc(100% / ${PIXELS_WIDE})`,
            }}
          />
        ))
      }
    </div>
  );
};

export default Avatar;
