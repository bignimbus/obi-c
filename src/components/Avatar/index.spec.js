import React from 'react';
import renderer from 'react-test-renderer';
import Avatar from '.';

const renderComponent = props => renderer.create(
  <Avatar />,
);

describe('Avatar', () => {
  test('default', () => {
    const element = renderComponent();
    expect(element.toJSON()).toMatchSnapshot();
  });
});
