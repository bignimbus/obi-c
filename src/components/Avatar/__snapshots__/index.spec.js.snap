// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar default 1`] = `
<div
  className="avatar"
>
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 0",
        "top": "calc(100% / 13 * 0)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 0",
        "top": "calc(100% / 13 * 1)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 0",
        "top": "calc(100% / 13 * 2)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 0",
        "top": "calc(100% / 13 * 3)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 0",
        "top": "calc(100% / 13 * 4)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 0",
        "top": "calc(100% / 13 * 5)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 0",
        "top": "calc(100% / 13 * 6)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 0",
        "top": "calc(100% / 13 * 7)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 0",
        "top": "calc(100% / 13 * 8)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 0",
        "top": "calc(100% / 13 * 9)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 0",
        "top": "calc(100% / 13 * 10)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 0",
        "top": "calc(100% / 13 * 11)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 0",
        "top": "calc(100% / 13 * 12)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 1",
        "top": "calc(100% / 13 * 12)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 2",
        "top": "calc(100% / 13 * 12)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 3",
        "top": "calc(100% / 13 * 12)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 4",
        "top": "calc(100% / 13 * 12)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 5",
        "top": "calc(100% / 13 * 12)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 6",
        "top": "calc(100% / 13 * 12)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 7",
        "top": "calc(100% / 13 * 12)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 8",
        "top": "calc(100% / 13 * 12)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 9",
        "top": "calc(100% / 13 * 12)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 10",
        "top": "calc(100% / 13 * 12)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 11",
        "top": "calc(100% / 13 * 12)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 11",
        "top": "calc(100% / 13 * 0)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 11",
        "top": "calc(100% / 13 * 1)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 11",
        "top": "calc(100% / 13 * 2)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 11",
        "top": "calc(100% / 13 * 3)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 11",
        "top": "calc(100% / 13 * 4)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 11",
        "top": "calc(100% / 13 * 5)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 11",
        "top": "calc(100% / 13 * 6)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 11",
        "top": "calc(100% / 13 * 7)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 11",
        "top": "calc(100% / 13 * 8)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 11",
        "top": "calc(100% / 13 * 9)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 11",
        "top": "calc(100% / 13 * 10)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 11",
        "top": "calc(100% / 13 * 11)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 1",
        "top": "calc(100% / 13 * 0)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 2",
        "top": "calc(100% / 13 * 0)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 3",
        "top": "calc(100% / 13 * 0)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 4",
        "top": "calc(100% / 13 * 0)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 5",
        "top": "calc(100% / 13 * 0)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 6",
        "top": "calc(100% / 13 * 0)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 7",
        "top": "calc(100% / 13 * 0)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 8",
        "top": "calc(100% / 13 * 0)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 9",
        "top": "calc(100% / 13 * 0)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 10",
        "top": "calc(100% / 13 * 0)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 4",
        "top": "calc(100% / 13 * 4)",
        "width": "calc(100% / 12)",
      }
    }
  />
  <div
    className="avatar__pixel"
    style={
      Object {
        "height": "calc(100% / 13)",
        "left": "calc(100% / 12 * 6",
        "top": "calc(100% / 13 * 4)",
        "width": "calc(100% / 12)",
      }
    }
  />
</div>
`;
